components/brave_shields/browser/domain_block_navigation_throttle.cc

```diff
@@ -164,6 +172,35 @@ void DomainBlockNavigationThrottle::OnShouldBlockDomain(
     // runner, but now we know that we want to allow navigation to continue.
     Resume();
     return;
+  } else if (new_url.is_valid()) {
+    content::NavigationHandle* handle = navigation_handle();
+
+    content::WebContents* contents = handle->GetWebContents();
+
+    // Cancel without an error status to surface any real errors during page
+    // load.
+    CancelDeferredNavigation(content::NavigationThrottle::ThrottleCheckResult(
+        content::NavigationThrottle::CANCEL));
+
+    content::OpenURLParams params =
+        content::OpenURLParams::FromNavigationHandle(handle);
+    params.url = new_url;
+    params.transition = ui::PAGE_TRANSITION_CLIENT_REDIRECT;
+    // We get a DCHECK here if we don't clear the redirect chain because
+    // technically this is a new navigation
+    params.redirect_chain.clear();
+
+    base::SequencedTaskRunnerHandle::Get()->PostTask(
```

Comment:
this approach to filtering is potentially suboptimal, but we can keep it as is as long the filter list consists of really small set of entries

Reply:
@iefremov What did you have in mind in term of a "really small set of entries"?

Here's an example list a user could add: https://github.com/AdguardTeam/FiltersRegistry/blob/master/filters/filter_17_TrackParam/filter.txt